src/EditorFeatures/CSharpTest/CodeActions/InvertIf/InvertIfTests.cs

```diff
@@ -639,5 +639,57 @@ void F()
 @"string x; if (x.Length <= 0.0f) { EqualsZero(); } else { GreaterThanZero(); }
     } } ");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElselessIfStatement_Foreach()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        foreach (var item in list)
+        {
+            [||]if (c)
+            f();
+        }
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        foreach (var item in list)
+        {
+            if (!c)
+                continue;
+            f();
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInvertIf)]
+        public async Task TestElselessIfStatement_Method()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        [||]if (c)
+        f();
+    }
+}",
+@"class C
+{
+    void M()
+    {
+        [||]if (!c)
```

Comment:
how possible this is passing..

Reply:
We remove the markup from both the before/after.  That's because we allow markup in the 'after' part to specify important pieces of data for some tests.

Reply:
nice. I'll use this to test each case in both direction. :D